### DISCUSSION TITLE
Proposal for project summary

### DISCUSSION THREAD

**Discussion by nyluntu** [sovelluskontti/project-semantic-search#87] [General]
Let's discuss about this on today.

**Target Audience:** Software developers seeking to gain familiarity with the concept of semantic search.

**Table of Contents**
1.	Introduction
2.	Theory/Background Information
3.	System Architecture
4.	Features
5.	Set up local enviroment
6.	Retrospective
7.	List of Sources

The project begins with an **Introduction** section, which outlines the **Purpose and Goals**—detailing the problem being addressed and the objectives the system aims to achieve. 

Following this, the **Theory/Background Information** provides foundational knowledge by explaining key concepts such as search engines, vector databases, semantic search, Docker, OpenAI, and embeddings. Additionally, any other relevant theoretical concepts necessary for understanding the project will be included. 

The **System Architecture** section offers a high-level overview, featuring a diagram and description of the technology stack used. 

The **Features** section describes the key functionalities and what the prototype is designed to accomplish. 

Next, the **Set up Local Environment** section provides detailed guidance on prerequisites, installation steps, configuration, and instructions on how to run the application, typically included in the project’s GitHub README file. 

In the **Retrospective section**, potential future features and identified limitations are discussed, along with any challenges or issues encountered during the development process.

Finally, the **List of Sources** section compiles all relevant references, external materials, and documentation that contributed to the development or provide additional helpful information.

Include a **video demonstration** showcasing how the application functions on your local machine. No need to explain every detail, just highlight the basic functionalities.

**Comment by nyluntu** [comment]
@sarafarahabadi I added some comments to the Word document. (Word review feature) Could you notify me when the summary document is ready so I can review it a bit more? Then we could discuss the scope of the thesis and what to include in it. If you can include a video showing how this application works, it would be great.

**Reply by sarafarahabadi** [reply]
Hello Turo, 
Thank you for your comments. I will apply them. I will let you know when the document is ready. I will record a video when the project summary is done and I will upload it to the oneDrive folder.

**Comment by sarafarahabadi** [comment]
I added the Summery to the OneDrive folder under the name "Sovelluskontti_Sara Farahabadi_FinalVersion". 
Please let me know if I have to add or edit anything. I will record the video tomorrow and add it there as well. 

Could you please provide me another work certificate like last time so I can upload it for university and have this internship graded? Thank you very much. 

About the thesis, Is it ok with you to have a meeting next Monday at 2 (the usual meeting time) to discuss the topic and its scope in detail?
I read the blog about image search and it seems really interesting to have three different search types in one project at the end. I would like to hear about your expectations of this thesis and what type of project you have in mind.

**Comment by sarafarahabadi** [comment]
Hello Turo, 

The recording is in the OneDrive folder. I recorded a detailed installment of the project covering all steps. When recording I realized I forgot a small step to mention in the summary so I fixed it also and uploaded the readMe file in GitHub and the new version under the name "FinalVersion_ProjectSummery_SaraFarahabadi". 

All tasks are done on the GitHub board as well. I wanted to clean and delete unnecessary files from the OneDrive but I couldn't since I am a guest contributor. The only files that are updated there, are the summary and video. The rest of the documents that I have uploaded can be deleted. 

Please let me know if there is anything I need to edit. Thank you. 
See you on Tuesday.
